### PR TITLE
Fixes 1848 : Option for adding location for those pictures which have no location

### LINKED ISSUE #1848: If no coordinates, ask where picture was taken
Often GPS does not work, or is not enabled, and the picture has no coordinates, making it probably useless, see for instance https://commons.wikimedia.org/wiki/File:%D8%B5%D9%88%D8%B1%D8%A9_%D9%84%D9%82%D8%B1%D9%8A%D8%AA%D9%8A_.jpg

In such cases, we should ask the user where the picture was taken.

### PR DESCRIPTION
**Pull request by Ayan-10**
**Description (required)**

Fixes #1848 

Some images don't have the location. It's very important to have a location for an upload.
1. Suggesting user to add location.
2. Map icon in the upload wizard is available for no location images too to add the location.
3. Skip option is also available.

**Tests performed (required)**

Tested latest master ProdDebug on Pixel 3 with API level 30.

### CODE REVIEW

**Comment by Ayan-10** [comment]
@nicolas-raoul In my local environment, this PR is building without any problem. But here I can see a grade build failure. This PR is up to date with the latest master.

**Comment by nicolas-raoul** [comment]
@Ayan-10 Sometimes CI does not work. Please do not worry too much about it, except if that happens for several commits in succession.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
It seems to generally work :-)

After uploading 2 pictures (with and without adding coordinates), the back button to exit the app worked strangely (thumbnails in my contributions move vertically and the app does not exit). Would you mind checking?

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -680,4 +680,7 @@ Upload your first media by tapping on the add button.</string>
   <string name="apply">Apply</string>
   <string name="reset">Reset</string>
   <string name="location_message">Location data helps Wiki editors find your picture, making it much more useful.\nYour recent uploads have no location.\nWe suggest you turn on location in your camera app\'s settings.\nThank you for uploading!</string>
+  <string name="no_location_found_title">No location found</string>
+  <string name="no_location_found_message">Location data helps Wiki editors find your picture, making it much more useful.\nThis image has no location. We suggest you add location for this image</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:684]: I would suggest: `How about adding the place where this image was taken? Location data helps Wiki editors find your picture, making it much more useful. Thank you!`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by nicolas-raoul** [comment]
Understood! Would you mind creating a new issue, if possible with reproduction steps? Thanks a lot!

**Comment by Ayan-10** [comment]
@nicolas-raoul The issue has been created.

**Comment by nicolas-raoul** [comment]
Would you mind rebasing? Thanks :-)

File: app/src/main/res/layout/fragment_upload_media_detail_fragment.xml [hunk 2]
```diff
@@ -127,8 +127,8 @@
                       android:layout_width="wrap_content"
                       android:layout_height="wrap_content"
                       android:layout_alignParentRight="true"
-                      android:enabled="false"
                       android:alpha=".5"
+                      android:enabled="false"
```
Comment (nicolas-raoul) [inline app/src/main/res/layout/fragment_upload_media_detail_fragment.xml:131]: Is this change needed?

**Review by nicolas-raoul — COMMENTED** [review]
It works great! Very useful addition.

Do you think it would be possible to add unit tests?

Also, would it be possible to remember the last used location? When uploading several pictures of the same place, having to scroll from San Francisco for every picture takes a lot of time. It can be a separate issue if you prefer.

File: app/src/main/res/layout/fragment_upload_media_detail_fragment.xml (hunk 2 above)
Comment (Ayan-10) [reply app/src/main/res/layout/fragment_upload_media_detail_fragment.xml:131]: No. I think this change is auto generated. I will revert it.

**Comment by Ayan-10** [comment]
Sure, I will add tests.

As, this is very related enhancement I will save the last location in this PR.

Thanks

File: app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java [hunk 3]
```diff
@@ -53,6 +54,7 @@
     UploadMediaDetailsContract.View, UploadMediaDetailAdapter.EventListener {
 
     private static final int REQUEST_CODE = 1211;
+    public static final String LAST_LOCATION = "last_location_while_uploading";
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java:57]: It works great! Please add some Javadoc and an example value, though. Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java (hunk 3 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java:60]: You should also explain that it is a key for applicationKvStore. Place yourself in the shoes of someone who sees this code for the first time and wonders what this String is for. The String itself obviously does not itself contain a location, but that's what the current Javadoc implies. Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Review by nicolas-raoul — APPROVED** [review]
